### PR TITLE
LWG Poll 8: P0943R6 Support C atomics in C++

### LINKED ISSUE #4332: [2020-11 LWG Motion 8] P0943R6 Support C atomics in C++
https://wiki.edg.com/pub/Wg21fall2020/StrawPolls/p0943r6.html

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #4332 
Fixes cplusplus/papers#365

### CODE REVIEW

File: source/atomics.tex [hunk 1]
```diff
@@ -3701,3 +3701,140 @@
...
+nor any of the non-macro global namespace declarations,
+are provided by any C++ standard library header
+other than \libheader{stdatomic.h}.
+
+\pnum
+\recommended
+Implementations should ensure
+that C and C++ representations of atomic objects are compatible,
+so that the same object can be accessed as both an \tcode{_Atomic(T)}
+from C code and an \tcode{atomic<T>} from C++ code.
```
Comment (burblebee) [inline source/atomics.tex:3837]: "C++"  ->  "\Cpp{}"

File: source/atomics.tex [hunk 2]
```diff
@@ -3701,3 +3701,140 @@
...
+\pnum
+Each of the \grammarterm{using-declaration}s for
+\tcode{int$N$_t}, \tcode{uint$N$_t}, \tcode{intptr_t}, and \tcode{uintptr_t}
+listed above is defined if and only if the implementation defines
+the corresponding typedef name in \ref{atomics.syn}.
+
+\pnum
+Neither the \tcode{_Atomic} macro,
+nor any of the non-macro global namespace declarations,
+are provided by any C++ standard library header
```
Comment (burblebee) [inline source/atomics.tex:3829]: "C++" -> "\Cpp{}"

**Review by burblebee — CHANGES_REQUESTED** [review]

File: source/atomics.tex (hunk 2 above)
Comment (johelegp) [reply source/atomics.tex:3829]: Looks like a tool check might be needed.

File: source/atomics.tex (hunk 1 above)
Comment (jensmaurer) [reply source/atomics.tex:3837]: fixed

File: source/atomics.tex (hunk 2 above)
Comment (jensmaurer) [reply source/atomics.tex:3829]: There are a lot of `extern "C++"` things floating around...

Comment (jensmaurer) [reply source/atomics.tex:3829]: ... fixed

**Review by burblebee — APPROVED** [review]
Thanks for fixing!  FYI, we only use the \Cpp macro when referring to the language or the spec; I don't know why it was added, given that we don't have a similar macro for C.

**Comment by johelegp** [comment]
It's because plain C++ doesn't look well rendered.

**Pull request merged**